sql: Add JNI-based Java Interface

StarostaGit opened the pull request:
`README.md` has been updated to include the proper way to run the build script for the java interface, producing a singular `.jar` file.

StarostaGit committed:
sql: Add JNI-based Java Interface

wjohnson commented on integration/sql/iface-java/src/java/io/openlineage/sql/SqlMeta.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.List;
+
+public class SqlMeta {
+    private final List<DbTableMeta> inTables;
```
There doesn't appear to be a public way of getting access to the input / output tables.

Shouldn't we have some getter or setter?

StarostaGit replied:
huh, weird, I had them locally. Maybe I accidentally deleted them resolving some conflict.
In any way, yes, they should absolutely be there.

wjohnson commented on integration/sql/iface-java/src/java/io/openlineage/sql/OpenLineageSql.java:
```diff
@@ -0,0 +1,15 @@
+    public static native String provider();
+    
+    static {
+        System.loadLibrary("openlineage_sql_java");
```
I have a very basic script that is importing the library but I'm receiving the following error works:

```text
Exception in thread "main" java.lang.UnsatisfiedLinkError: no openlineage_sql_java in java.library.path
        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1860)
        at java.lang.Runtime.loadLibrary0(Runtime.java:871)
        at java.lang.System.loadLibrary(System.java:1124)
        at io.openlineage.sql.OpenLineageSql.<clinit>(OpenLineageSql.java:13)
        at App.main(App.java:15)
```

I'm compiling the below script with `javac -cp ".:../path/to/integration/sql/iface-java/target/openlineage-sql-java-1.0-SNAPSHOT.jar" App.java` and executing it similarly with `java -cp ".:../path/to/integration/sql/iface-java/target/openlineage-sql-java-1.0-SNAPSHOT.jar" App.java`. 

I can see the so file in `/integration/sql/target/debug/` but I'm not sure how to include it in the class path.  Shouldn't it get bundled with the jar file itself?

```java
import io.openlineage.sql.OpenLineageSql;
import io.openlineage.sql.SqlMeta;

import java.util.ArrayList;
import java.util.Arrays;

public class App {
    public String getGreeting() {
        return "Hello World!";
    }

    public static void main(String[] args){
        System.out.println(new App().getGreeting());
        SqlMeta output = OpenLineageSql.parse(Arrays.asList("SELECT * FROM test"));
        System.out.println(output.toString());
    }
}
```

StarostaGit replied:
Looking into it

StarostaGit replied:
The new commit should fix that

wjohnson reviewed (changes requested):
It would be great to revisit the public accessor methods for SqlMeta and to at least share how you are supposed to point to / include the rust library in a java app.

StarostaGit committed:
Feedback

Added getters to `SqlMeta` class.
Reworked java lib packaging to add the native implementation to the resulting
jar file and link against it.
Added an integration test, which also provides a way to manually test
the java interface.

StarostaGit committed:
Improved build scripts
